suite-native/module-authorize-device/src/screens/connect/TurnOnAndUnlockDeviceScreen.tsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useFocusEffect, useNavigation } from '@react-navigation/native';
+
+import { useAlert } from '@suite-native/alerts';
+import {
+    selectBluetoothAdapterStatus,
+    selectHasKnownBluetoothDevices,
+    selectUnknownNearbyBluetoothDevices,
+    useBluetoothManager,
+} from '@suite-native/bluetooth';
+import { TurnOnAndUnlockDeviceScreenContent } from '@suite-native/device';
+import { Translation } from '@suite-native/intl';
+import {
+    AuthorizeDeviceStackParamList,
+    AuthorizeDeviceStackRoutes,
+    Screen,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+import { isAndroid } from '@trezor/env-utils';
+import { TimerId } from '@trezor/type-utils';
+
+import { BluetoothPairingHelpButton } from '../../components/connect/BluetoothPairingHelpButton';
+import { BluetoothPairingHints } from '../../components/connect/BluetoothPairingHints';
+import { BluetoothPairingSettings } from '../../components/connect/BluetoothPairingSettings';
+import { ConnectDeviceScreenHeader } from '../../components/connect/ConnectDeviceScreenHeader';
+
+type NavigationProps = StackNavigationProps<
+    AuthorizeDeviceStackParamList,
+    AuthorizeDeviceStackRoutes.TurnOnAndUnlockDevice
+>;
+
+let timeoutId: TimerId;
```

Comment:
It would be safer to use ref for the timeout reference. This can (theoretically) cause race conditions if multiple instances of this screen are mounted. It might happen during development with hot reloading.

Reply:
Fixed in 88fa3d6118559895cc2804cc87721f4790ddd9e0.